code/game/machinery/computer/atmos_computers/_air_sensor.dm

```diff
@@ -15,6 +15,53 @@
 	id_tag = chamber_id + "_sensor"
 	return ..()
 
+/obj/machinery/air_sensor/Destroy()
+	. = ..()
+	reset()
+
 /obj/machinery/air_sensor/update_icon_state()
 	icon_state = "gsensor[on]"
 	return ..()
+
+/obj/machinery/air_sensor/proc/reset()
+	if(GLOB.objects_by_id_tag[chamber_id + "_in"] != null)
+		var/obj/machinery/atmospherics/components/unary/outlet_injector/injector = GLOB.objects_by_id_tag[chamber_id + "_in"]
+		injector.disconnect_chamber()
+
+	if(GLOB.objects_by_id_tag[chamber_id + "_out"] != null)
+		var/obj/machinery/atmospherics/components/unary/vent_pump/pump  = GLOB.objects_by_id_tag[chamber_id + "_out"]
+		pump.disconnect_chamber()
+
+
+///right click with multi tool to disconnect everything
+/obj/machinery/air_sensor/multitool_act_secondary(mob/living/user, obj/item/tool)
+	to_chat(user,"You reset all I/O ports")
+	reset()
+	return TRUE
+
+/obj/machinery/air_sensor/multitool_act(mob/living/user, obj/item/multitool/I)
+	.= ..()
+
+	if (istype(I))
+		if(istype(I.buffer, /obj/machinery/atmospherics/components/unary/outlet_injector))
+			var/obj/machinery/atmospherics/components/unary/outlet_injector/injector = I.buffer
+			injector.chamber_id = chamber_id
+			GLOB.objects_by_id_tag[chamber_id + "_in"] = injector
```

Comment:
We repeat this `+ "_in"` thing a lot, I know because I remember futzing with it a few times. Can you make this its own macro in the atmos defines file? Something like:

```dm
#define CHAMBER_INPUT_FROM_ID(chamber_id) (chamber_id + "_in")
#define CHAMBER_OUTPUT_FROM_ID(chamber_id) (chamber_id + "_out")
```

...then use that everywhere?

Reply:
I'll do that